test/e2e/fixtures/when.go

```diff
@@ -222,14 +222,18 @@ func (w *When) WaitForWorkflow(options ...interface{}) *When {
 		select {
 		case event := <-watch.ResultChan():
 			wf, ok := event.Object.(*wfv1.Workflow)
-			print(".")
 			if ok {
 				w.hydrateWorkflow(wf)
 				if condition(wf) {
 					println("Condition met after", time.Since(start).Truncate(time.Second).String())
 					w.wf = wf
 					return w
 				}
+				// once done the workflow is done, the condition can never be met
```

Comment:
this reduces the chances we'll get test suites that fail, but only after a long time